server/monitor.go

```diff
@@ -2686,9 +2688,98 @@ func (s *Server) HandleJsz(w http.ResponseWriter, r *http.Request) {
 	}
 	b, err := json.MarshalIndent(l, "", "  ")
 	if err != nil {
-		s.Errorf("Error marshaling response to /leafz request: %v", err)
+		s.Errorf("Error marshaling response to /jsz request: %v", err)
 	}
 
 	// Handle response
 	ResponseHandler(w, r, b)
 }
+
+type HealthStatus struct {
+	Status string `json:"status"`
+	Error  string `json:"error,omitempty"`
+}
+
+// https://tools.ietf.org/id/draft-inadarei-api-health-check-05.html
+func (s *Server) HandleHealthz(w http.ResponseWriter, r *http.Request) {
```

Comment:
I assume this needs exposing via the server status requests in events.go line 859 and become part of the table monSrvc.

Provided that, this probably has to be split into two functions and have returns that align with similar calls.

Reply:
I don't think so, this one is really only http and for the environments like k8s and liveness probes etc.
I could be wrong though and we could then put it there but does not feel like same level as the others IMO.

Reply:
Exposing as a NATS endpoint would be useful for future NATS tooling around server maintenance.

Reply:
Longer discussion but there is no value there IMO.